src/System.ServiceProcess.ServiceController/tests/ServiceBaseTests.cs

```diff
@@ -75,104 +76,125 @@ public void Cleanup()
         [ConditionalFact(nameof(IsProcessElevated))]
         public void TestOnStartThenStop()
         {
-            var controller = new ServiceController(_testService.TestServiceName);
-            AssertExpectedProperties(controller);
-            string expected =
-@"OnStart args=
-OnStop
-";
-            controller.Stop();
-            controller.WaitForStatus(ServiceControllerStatus.Stopped);
-            Assert.Equal(expected, _testService.GetServiceOutput());
+            string serviceName = _testService.TestServiceName;
+            using (var client = new NamedPipeClientStream(".", serviceName, PipeDirection.In))
```

Comment:
I think it would be worth keeping most of the duplicated logic about creating, connecting, reading the pipe encapsulated in the TestService class.

Reply:
The tests and Testservice run as different processes. So we cant have both client and server named pipes in the testservice.

Reply:
I'm just talking about the class we are using here in our tests the _testService object.

Reply:
Okay yeah. I think we should be able to  do that , I will just try that. Thanx for pointing out